File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& v) {
        int ans=0;
        int n=v.size();
        int mn=v[0];
        for(int i=1;i<n;i++){
            int profit=v[i]-mn;
            ans=max(ans,profit);
            mn=min(mn,v[i]);
        }
        return ans;
    }
};

File: 121-best-time-to-buy-and-sell-stock/best-time-to-buy-and-sell-stock.py
class Solution:
    def maxProfit(self, nums: List[int]) -> int:
        n=len(nums)
        a=nums[0]
        ans=0
        for i in range(1,n):
            a=min(nums[i],a)
            ans=max(ans,nums[i]-a)
        return ans
